src/Runner.Worker/ExecutionContext.cs

```diff
@@ -437,6 +442,15 @@ public TaskResult Complete(TaskResult? result = null, string currentOperation =
 
             PublishStepTelemetry();
 
+            StepResult.Conclusion = _record.Result ?? TaskResult.Succeeded;
```

Comment:
maybe just a private `var stepResult` since you never directly consume it from ExecutionContext?

Reply:
I was using it from other place, but I think I don't need it anymore